### PR TITLE
fix #9096 feat(nimbus): Add inclusion and exclusion targeting APIs

### LINKED ISSUE #9043: Inclusion/Exclusion Targeting UI
Let’s add some UI to the Audience page in Experimenter that has two drop downs, one called ‘Include Experiments’ and one called ‘Exclude Experiments’. Each drop down should be a multiselect that lists all NimbusExperiments in all statuses. A user can select any number of other experiments in each list, but no experiment should be selectable in both. The selected included and excluded experiments should appear as part of the targeting JEXL string in the recipe as either being present or not present in the list of enrolled experiment slugs in the client.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-3643)

### LINKED ISSUE #9096: API changes for inclusion/exclusion targeting
┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-3673)

### PR DESCRIPTION
**Pull request by brennie**
Because:
- we want to target against past and present enrollments in other
  experiments and rollouts

this commit:
- updates the API with `excluded_experiments` and `required_experiments`
  fields; and
- adds a new gql query to list experiments filtered by application
  to allow the front-end to populate these fields in an upcoming change.

### CODE REVIEW

**Comment by jaredlockhart** [comment]
We should either block this PR or at least version gate it on the versions this will be available in the client, which might not be clear since the patches aren't landed yet.

**Comment by jaredlockhart** [comment]
Oh yeah you've got a version gate in there, but we should at least wait until we know what the version is before we land this.  

Also sidebar, I would've split this apart into at least backend and frontend changes rather than lumping it all together.  We should always expect that a PR is revertible in prod, with the caveat that migrations aren't easily reverted, so isolating changes into smaller prs as much as possible should be preferred.

But everything in here looks generally correct to me 👍

**Comment by brennie** [comment]
Blocked waiting for mobile to support `enrollments` targeting variable

File: experimenter/experimenter/experiments/api/v5/serializers.py
Comment (jaredlockhart) [inline experimenter/experimenter/experiments/api/v5/serializers.py:1549]: I love sets so much 🙏

File: experimenter/experimenter/experiments/api/v5/types.py
Comment (jaredlockhart) [inline experimenter/experimenter/experiments/api/v5/types.py:677]: Slug and not name?  Name might be more humane?

Comment (jaredlockhart) [reply experimenter/experimenter/experiments/api/v5/types.py:677]: Also above it says it's a list of ExperimentType but here we're only providing id and slug, would it make sense to add a type that's just those two fields?

File: experimenter/experimenter/experiments/api/v5/queries.py
Comment (jaredlockhart) [inline experimenter/experimenter/experiments/api/v5/queries.py:41]: Why `meta` in the name here?  I would expect this to be called something like `experiments_by_application`?

**Review by jaredlockhart — COMMENTED** [review]
Looking good.  I would still split this by backend/frontend changes to be smaller to review/land.

File: experimenter/experimenter/experiments/api/v5/types.py
Comment (brennie) [reply experimenter/experimenter/experiments/api/v5/types.py:677]: The .only() is just a DB optimization because we know in practice the query is just ehse fields. If the query asks for more fields, the model will query for them on-the-fly.

Comment (brennie) [reply experimenter/experimenter/experiments/api/v5/types.py:677]: Also turns out the query was asking for name and publicDescription already, so I've added those to the query and will update this soon.

File: experimenter/experimenter/experiments/tests/api/v5/test_queries.py
Comment (jaredlockhart) [inline experimenter/experimenter/experiments/tests/api/v5/test_queries.py:2228]: Probably `TestNimbusExperimentsByApplicationQuery` now?

File: experimenter/experimenter/experiments/tests/api/v5/test_queries.py
Comment (jaredlockhart) [inline experimenter/experimenter/experiments/tests/api/v5/test_queries.py:2235]: You can probably get rid of this test altogether, the other one tests both the inclusion and exclusion cases?

File: experimenter/experimenter/experiments/tests/api/v5/test_serializers/test_nimbus_experiment_serializer.py
Comment (jaredlockhart) [inline experimenter/experimenter/experiments/tests/api/v5/test_serializers/test_nimbus_experiment_serializer.py:1501]: I don't think that's necessary, just explicitly set both fields in the test.

**Review by jaredlockhart — APPROVED** [review]
Yep looks great, thnx for splitting it up.  I don't think you need to parameterize the tests across both fields, I'd just hardcode them, but up to you.  Thnx @brennie ! 🎉

**Pull request merged**
